cmd/native-relayer/cli/start.go

```diff
@@ -0,0 +1,147 @@
+package cli
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/block-vision/sui-go-sdk/signer"
+	"github.com/block-vision/sui-go-sdk/sui"
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/gonative-cc/relayer/dal"
+	"github.com/gonative-cc/relayer/env"
+	"github.com/gonative-cc/relayer/ika"
+	"github.com/gonative-cc/relayer/ika2btc"
+	"github.com/gonative-cc/relayer/native"
+	"github.com/gonative-cc/relayer/native2ika"
+	"github.com/gonative-cc/relayer/nbtc"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+var startCmd = &cobra.Command{
+	Use:   "start",
+	Short: "Starts the relayer",
+	Run: func(cmd *cobra.Command, args []string) {
+		lvl, err := cmd.Root().PersistentFlags().GetString("log-level")
+		if err != nil {
+			log.Error().Err(err).Msg("Error getting log level")
+			os.Exit(1)
+		}
+		logLvl, err := zerolog.ParseLevel(lvl)
+		if err != nil {
+			log.Error().Err(err).Msg("Error parsing log level")
+			os.Exit(1)
+		}
+		env.InitLogger(logLvl)
+		configFile, err := cmd.Root().PersistentFlags().GetString("config")
+		if err != nil {
+			log.Error().Err(err).Msg("Error getting config file path")
+			os.Exit(1)
+		}
+
+		config, err := loadConfig(configFile)
+		if err != nil {
+			log.Error().Err(err).Msg("Error loading config")
+			os.Exit(1)
+		}
+		db, err := dal.NewDB(config.DB.File)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating database")
+			os.Exit(1)
+		}
+		if err := db.InitDB(); err != nil {
+			log.Error().Err(err).Msg("Error initializing database")
+			os.Exit(1)
+		}
+		suiClient := sui.NewSuiClient(config.Ika.RPC).(*sui.Client)
+		if suiClient == nil {
+			log.Error().Err(err).Msg("Error creating Sui client")
+			os.Exit(1)
+		}
+		signer, err := signer.NewSignertWithMnemonic(config.Ika.SignerMnemonic)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating signer with mnemonic")
+			os.Exit(1)
+		}
+		ikaClient, err := ika.NewClient(
+			suiClient,
+			signer,
+			ika.SuiCtrCall{
+				Package:  config.Ika.NativeLcPackage,
+				Module:   config.Ika.NativeLcModule,
+				Function: config.Ika.NativeLcFunction,
+			},
+			ika.SuiCtrCall{
+				Package:  config.Ika.NativeLcPackage,
+				Module:   config.Ika.NativeLcModule,
+				Function: config.Ika.NativeLcFunction,
+			},
+			config.Ika.GasAcc,
+			config.Ika.GasBudget,
+		)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating IKA client")
+			os.Exit(1)
+		}
+		btcProcessor, err := ika2btc.NewProcessor(
+			rpcclient.ConnConfig{
+				Host:         config.Btc.RPCHost,
+				User:         config.Btc.RPCUser,
+				Pass:         config.Btc.RPCPass,
+				HTTPPostMode: config.Btc.HTTPPostMode,
+				DisableTLS:   config.Btc.DisableTLS,
+				Params:       config.Btc.Network,
+			},
+			config.Btc.ConfirmationThreshold,
+			db,
+		)
```

Comment:
same for other parts, create a helper function, which will take part of the config

Reply:
Done